File: mobile/src/main/java/io/ashdavies/cinnamon/inject/ApplicationScope.java
package io.ashdavies.cinnamon.inject;

import dagger.releasablereferences.CanReleaseReferences;
import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import javax.inject.Scope;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Scope
@Documented
@Retention(RUNTIME)
@CanReleaseReferences
public @interface ApplicationScope {
}

File: mobile/src/main/kotlin/io/ashdavies/cinnamon/inject/ApplicationScope.kt
package io.ashdavies.cinnamon.inject

import dagger.releasablereferences.CanReleaseReferences
import javax.inject.Scope
import kotlin.annotation.AnnotationRetention.RUNTIME

@Scope
@Retention(RUNTIME)
@CanReleaseReferences
annotation class ApplicationScope
